tsdb/head.go

```diff
@@ -1179,7 +1218,7 @@ func (a *initAppender) Append(ref uint64, lset labels.Labels, t int64, v float64
 
 func (a *initAppender) AppendExemplar(ref uint64, l labels.Labels, e exemplar.Exemplar) (uint64, error) {
 	// Check if exemplar storage is enabled.
-	if a.head.opts.NumExemplars <= 0 {
+	if !a.head.opts.EnableExemplarStorage && a.head.opts.MaxExemplars > -1 {
```

Comment:
I guess you meant this? (but look at my other comment about using `opts.MaxExemplars`)
```suggestion
	if !a.head.opts.EnableExemplarStorage || a.head.opts.MaxExemplars < 0 {
```

Reply:
No I mean `&&`, the storage must be explicitly enabled and the MaxExemplars needs to be non-negative.

Reply:
At the moment, it will add into buffer if exemplars is disabled with negative max exemplars

Reply:
:facepalm: